src/psyclone/psyGen.py

```diff
@@ -2255,15 +2441,22 @@ def update(self):
         '''
         from fparser.common.readfortran import FortranStringReader
         from fparser.two.Fortran2003 import Comment
+
+        # Ensure the fparser2 AST is up-to-date for all of our children
+        Node.update(self)
+
         # Check that we haven't already been called
         if self._ast:
             return
+
         # Find the locations in which we must insert the begin/end
         # directives...
         # Find the children of this node in the AST of our parent node
         try:
-            start_idx = self._parent._ast.content.index(self._children[0]._ast)
-            end_idx = self._parent._ast.content.index(self._children[-1]._ast)
+            start_idx = object_index(self._parent._ast.content,
```

Comment:
pylint ...

W:2456,37: Access to a protected member _ast of a client class (protected-access)
W:2457,37: Access to a protected member _ast of a client class (protected-access)
W:2458,35: Access to a protected member _ast of a client class (protected-access)
W:2459,35: Access to a protected member _ast of a client class (protected-access)

Reply:
Fixed.